homeassistant_api/models/domains.py

```diff
@@ -55,7 +56,14 @@ def __init__(
     )
 
     @classmethod
-    def from_json(
+    @override
+    def from_json(cls, json: Union[dict[str, JSONType], Any, None], **kwargs) -> Self:
+        raise ValueError(
+            f"`{cls.__name__}` does not support `from_json()`. Use `from_json_with_client()`"
```

Comment:
This is a nifty idea to rename it to `from_json_with_client` making the dependence on a client more explicit. I like it.